### DISCUSSION TITLE
Ckan patch

### DISCUSSION THREAD

**Discussion by DeepakSinghSurya** [ckan/ckan#6130] [Q&A]
How to play with CKan patch functionality what is the perfect way to write curl for ckan patch

**Comment by anuveyatsu** [comment]
@DeepakSinghSurya your question is not very clear to me but I guess you want to patch datasets in the CKAN instance. To do so you could use a CLI tool instead of curl. Here is the github repo - https://github.com/ckan/ckanapi

**Comment by DeepakSinghSurya** [comment]
Hi Anuar,
Thanks for the reply.
I agree with you, My question was not clear I wanted to understand will be
possible to patch "extras[]" using the package patch, if yes !  is there
any documentation on that?

On Thu, Jun 10, 2021 at 3:14 PM Anuar Ustayev ***@***.***>
wrote:

> @DeepakSinghSurya <https://github.com/DeepakSinghSurya> your question is
> not very clear to me but I guess you want to patch datasets in the CKAN
> instance. To do so you could use a CLI tool instead of curl. Here is the
> github repo - https://github.com/ckan/ckanapi
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ckan/ckan/discussions/6130#discussioncomment-850254>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMCQG4A5KD3R25UEK6YU5SDTSCCQZANCNFSM456ENGPA>
> .
>


-- 
[image: photo]
Deepak Singh
  ***@***.***>

Create your own email signature
<https://www.wisestamp.com/signature-in-email/?utm_source=promotion&utm_medium=signature&utm_campaign=create_your_own>

**Reply by anuveyatsu** [reply]
@DeepakSinghSurya yes, you can patch any metadata attribute using `package_patch` API. The docs are here:

https://docs.ckan.org/en/2.9/api/index.html#ckan.logic.action.patch.package_patch

**Comment by DeepakSinghSurya** [comment]
Awesome, thanks!
I try to utilize the package_patch API for "extras[]" (key-value
par section) with following syntax and it did not work
Ex :

package_patch?extras=[{"value": "elit voluptate","key":
"alignment_workflow_submitter_id"}, {"value": "elit voluptate","key":
"alignment_workflow_submitter_ids"}]



"extras": [
{
"value": "xx",
"key": "yy"
},
{
"value": "xx",
"key": "yy"
},
{
"value": "elit voluptate",
"key": "alignment_workflow_submitter_id"
},

On Fri, Jun 11, 2021 at 1:01 PM Anuar Ustayev ***@***.***>
wrote:

> @DeepakSinghSurya <https://github.com/DeepakSinghSurya> yes, you can
> patch any metadata attribute using package_patch API. The docs are here:
>
>
> https://docs.ckan.org/en/2.9/api/index.html#ckan.logic.action.patch.package_patch
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ckan/ckan/discussions/6130#discussioncomment-856929>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMCQG4GMFUYMCFPX5QDS5LDTSG3TTANCNFSM456ENGPA>
> .
>


-- 
[image: photo]
Deepak
Singh at  HCL Technologies
A  No:40, Kalayan AECS ,3rd Main, 2nd Cross,Bangalore-94
P  9916392480  <9916392480>
E  ***@***.***  ***@***.***>

Create your own email signature
<https://www.wisestamp.com/signature-in-email/?utm_source=promotion&utm_medium=signature&utm_campaign=create_your_own>
